R/get_covid_data.R

```diff
@@ -0,0 +1,25 @@
+#' Get COVID data from German Nowcast Hub
+#'
+#' @param url String of url for raw data
+#' @param loc_to_subs String indicating the location to subset the data to,
+#'    default is "DE" which will get the national data for all age groups.
+#' @return Data.frame of counts of cases by reference date and report date for
+#'    only Germany and
+get_covid_data <- function(url,
+                           loc_to_subset = "DE") {
+  raw_data <- readr::read_csv(config$covid$url) |>
+    rename(value_81d = `value_>80d`) # Not sure if this is really correct but otherwise get NAs
```

Comment:
@seabbs I was trying to find how you handled this but couldn't. I figured this would just make the delay distrib have a weird upward tale on day 81 but should otherwise work (though in this example, our max delay is 40 so it wouldn't capture this)

Reply:
I'm not sure I totally understand. The max delay we used was 40 for the baseline model and 28 for the epinowcast model here so I think it doesn't quite come up? 

More generally yes you can do this and it does result in a bump at the tail. It can be handled using a non-parametric adjustment

Reply:
Yeah I just wanted to handle it since we will be modifying to look at longer delays (though I think as written max would be 200% so 80, so it still doesn't matter)

Reply:
Yeah I edited this to use the August 8th, 2022 dataset. 

However, we had decided to use a 80 day rolling window eval dataset (this was discussed by them after the fact as a more fair eval). 

I am now thinking we should do the rolling window eval for norovirus and for the model permutation analysis, but specifically for the Hub validation we will probably want to use exactly what they used (August 8th, 2022) dataset. 

Thoughts? I'll have to separate the scoring objects then bc they will have different eval datasets, but I think thats fine.

Reply:
I think we want to use the same thing for the validation and the model config. I think that we want to track their secondary analysis which I think was this rolling data idea but we need to check that rather than their main analysis in all cases as they and we agree it is the better choice. We need to make sure we say this in the paper. 

For the validation against noro I think we should follow exactly what they did in the same way we have with other choices. We can add something to the discussion about limitations for this if we wish. We should also make sure we clearly say what the evaluation data is if we are not.

Reply:
IMO this sorts of discussions should be in the paper doc so we can keep track and then move to action here. i.e the code follows the paper not the paper follows the code or some mix